src/runtime-rs/crates/runtimes/common/src/types/mod.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+mod trans_from_shim;
+mod trans_into_shim;
+
+use std::fmt;
+
+use anyhow::{Context, Result};
+use kata_sys_util::validate;
+use kata_types::mount::Mount;
+
+/// Request: request from shim
+/// Request and Response messages need to be paired
+#[derive(Debug, Clone)]
+pub enum Request {
+    CreateContainer(ContainerConfig),
+    CloseIo(ProcessID),
+    DeleteProcess(ProcessID),
+    Exec(ExecProcessRequest),
+    KillProcess(KillRequest),
+    Wait(ProcessID),
+    StartProcess(ProcessID),
+    StateProcess(ProcessID),
+    Shutdown(ShutdownRequest),
+    Pause(String),
+    Resume(String),
+    ResizePty(ResizePtyRequest),
+    Stats(String),
+    Update(UpdateRequest),
+    Pid,
+    Connect(String),
+}
+
+impl fmt::Display for Request {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+/// Response: response to shim
+/// Request and Response messages need to be paired
+#[derive(Debug, Clone)]
+pub enum Response {
+    CreateContainer(ShimPid),
+    CloseIo,
+    DeleteProcess(ProcessStateInfo),
+    Exec,
+    KillProcess,
+    Wait(ProcessExitStatus),
+    StartProcess(ShimPid),
+    StateProcess(ProcessStateInfo),
+    Shutdown,
+    Pause,
+    Resume,
+    ResizePty,
+    StatsResponse(StatsInfo),
+    Update,
+    Pid(ShimPid),
+    Connect(ShimPid),
+}
+
+impl std::fmt::Display for Response {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", &self)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ProcessType {
+    Container,
+    Exec,
+}
+
+#[derive(Clone, Debug)]
+pub struct ProcessID {
+    pub cid: String,
```

Comment:
As stated above, I'd like having a `ContainerID` type. Also keep naming conventions consistent, use `container_id` to match `process_type`.

Reply:
Thanks for your review, already `ContainerID` but for ProcessID I think it's unnecessary to use `ContainerID`.

Reply:
Thanks for your review, referring to all your suggestions, it is indeed clearer to use ContainerID in ProcessID, and use `container_id, exec_id` instead of `cid, eid`.